conversion.c

```diff
@@ -83,23 +75,27 @@ char* Hexadecimal (int num) {
 	char* storeBin = Binary (num);                  // Stores the binary value of the string returned
 	if (storeBin == NULL) return NULL;
 	for (int i = 0; storeBin[i] != '\0'; i++)length++;
-	char* hexString = malloc (sizeof (char) * ((length / 4) + 1));
-	if (hexString == NULL)exit (ERROR_MEM_ALLOC_FAILURE);
-	while ((binLen < length) && (hexLen < (length / 4))) {
+	char* hexStr = malloc (sizeof (char) * ((length / 4) + 1));
+	if (hexStr == NULL)exit (ERROR_MEM_ALLOC_FAILURE);
+	while (binLen < length && hexLen < (length / 4)) {
 		sum = 0;
 		for (int i = 3; i >= 0; i--, binLen++) {
 			sum += (storeBin[binLen] - '0') * pow (2, i);
 		}
-		hexString[hexLen] = Arr[sum];
+		hexStr[hexLen] = Arr[sum];
 		hexLen++;
 	}
-	hexString[hexLen] = '\0';
-	return hexString;
+	hexStr[hexLen] = '\0';
+	return hexStr;
+}
+
+/// <summary> Method for printing hexadecimal characters using type specifier </summary>
+void Hex (int num) {
+	printf ("HEX : %X\n", num);
 }
 
 /// <summary> Prints the output of the program </summary>
 int Call (int num, char* bin, char* hex) {
-	;
 	if (bin == NULL) {
 		printf ("The input exceeds the maximum length (32 bit)\n");
```

Comment:
This check for exceeding 32 bits does not work for some values.
Example 2 raised to power 33 is greater a 32 bit number but this message is not printed, instead it calculated the binary for the maximum value it can.